ambry-protocol/src/main/java/com.github.ambry.protocol/ReplicationControlAdminRequest.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.protocol;
+
+import com.github.ambry.utils.Utils;
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+
+/**
+ * An admin request used to control replication behavior (enable/disable)
+ */
+public class ReplicationControlAdminRequest extends AdminRequest {
```

Comment:
`AdminRequest` was written for  trigger compaction. The code is being reused here with minimal changes (if any) to avoid cascading changes. `AdminRequest` is not an abstract class and some requests (like trigger compaction) can be made without any extra information other than the partition id - so the partition id is in `AdminRequest`

I don't understand the question about `RequestOrResponse` version - that change wasn't made in this PR but the reason to have a version there is because that format can be changed and all of `PutRequest`, `GetRequest` and all other request/responses use the code in `RequestOrResponse` as a header (client id, correlation id etc). `AdminRequest` also has its own version.

Multiple partition ids is not required support. The API is made simple (and was written for trigger compaction and reused here). If you look at the `ServerAdminTool`, it supports disabling multiple partitions at once. These are rare requests - batching them is not going to improve performance by a whole lot but just ends up complicating the code (how do you return status if all but one succeeded?) 

We can introduce requests for all of what you suggest but that would blow up this change and this is the first step to doing all of that anyway. I will look to introduce metrics to track that but I think having APIs for all of that is excessive. Each API is a lot of classes (since we are not doing http) so unless its absolutely required, I'm not for introducing them.

I'll add java docs about what happens on restart - that will be useful.